File: gen_scripts/python2/k10.parameters.py
#!/usr/bin/env python2

# create the subroutine
def addNums (*numbers):
   sum = 0                            # initialize starting sum value
   for num in numbers:                # collection loop to get each number
      sum += num                      # sum up nums
   print "The summation is: %d." % sum # output results


print "Sending: 5, 2, 4, 3, 6"
# call the subroutine
addNums(5, 2, 4, 3, 6)   # pass a series of numbers

File: gen_scripts/ruby/k10.parameters.rb
#!/usr/bin/env ruby

# create method (subroutine)
def addNums (*numbers)
   sum = 0                                     # initialize starting sum value
   puts "Sending: #{numbers.join(", ")}"       # output numbers being sent
   numbers.each {|num| sum += num }            # iterate and sum up nums
   puts "The summation is: #{sum}."            # output results
end

# call the method (subroutine)
addNums 5, 2, 4, 3, 6
